backend/src/hooks/check-has-permission/check-has-permission.hook.ts

```diff
@@ -0,0 +1,34 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+import { HttpCode, PermissionKey } from '~/common/enums/enums';
+import { UsersByIdResponseDto } from '~/common/types/types';
+import { PermissionsError } from '~/exceptions/exceptions';
+
+const checkHasPermissions =
+  (permissions: PermissionKey[]) =>
```

Comment:
Sorry, it was my fault, please return this back
```suggestion
  (...permissions: PermissionKey[]) =>
```